### PR TITLE
Journal API: Add actor starting call

### PR DESCRIPTION
**Pull request by chbatey**
Refs #25970

### CODE REVIEW

File: akka-persistence/src/main/scala/akka/persistence/journal/AsyncRecovery.scala [hunk 1]
```diff
@@ -4,15 +4,30 @@
...
+   * Plugin API: optional API to be notified that a persistent actor is starting. Recovery
+   * of the actor won't start until the returned future has completed.
+   *
+   * @param persistenceId persistent actor id
+   * @param ref persistent actor ref
+   *
+   * This call is protected with a circuit-breaker.
+   */
+  @silent("never used")
+  def asyncPersistentActorStarting(persistenceId: String, ref: ActorRef): Future[Done] = Future.successful(Done)
```
Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/journal/AsyncRecovery.scala]: We need to verify if it's possible to add a new method to a trait like this in a compatible way. It wasn't in 2.11.

Plugin compiled with older Akka version. Running with newer Akka version that calls this method.

If it doesn't work can add an additional trait for the new method and check if the plugin implements that trait.

Comment (chbatey) [reply akka-persistence/src/main/scala/akka/persistence/journal/AsyncRecovery.scala]: @patriknw do you tink we should backport this? If so it'll need to support 2.11. 
If we don't it is a long time until the plugins can use this, but that may be okay.

**Comment by chbatey** [comment]
PLS BUILD

File: akka-persistence/src/main/scala/akka/persistence/journal/AsyncRecovery.scala (hunk 1 above)
Comment (patriknw) [reply akka-persistence/src/main/scala/akka/persistence/journal/AsyncRecovery.scala]: Will akka-persistence-cassandra 1.0 be for Akka 2.5 and 2.6, or only 2.6?
Akka 2.5 is officially supported for 1 more year.

**Comment by octonato** [comment]
This PR seems to be forgotten. In the meantime akka-persistence-cassandra 1.0 is out and only supporting 2.6.

**Pull request closed without merging**